### PR TITLE
feat(ollamy): stripe product payment manager by user

### LINKED ISSUE #233: [OLL-357] Backend - Billing Page route and subscription
As a : user of Ollamy

I want : to be able to create a subscription or payment system of my own.

Description : From the stripe payment system, you can create either payment links or subscription systems, which can be modified from certain endpoints. And potentially, depending on stripe's restrictions, from a users' panel.

Definition Of Done :

- [X] See how the stripe payment link system is available and implement it if possible ;
- [X] Implement a subscription system that renews on a monthly or weekly basis ;
- [X] Creation of the endpoint system for payment links, this endpoint will return a link and will take a user id as parameters ;
- [X] Creation of the enpdoint system for subscriptions, the user will have to choose his types among the different subscriptions created from the SLOTS Systems tasks.

Estimated Charge : 4 JEH

* Yoël EDERY (4 JEH)

<sub>From [SyncLinear.com](https://synclinear.com) | [OLL-357](https://linear.app/ollamy/issue/OLL-357/backend-billing-page-route-and-subscription)</sub>

### PR DESCRIPTION
**Pull request by ThisisYoYoDev**
## Summary

Please write a summary of your changes. Provide any information that could be
useful to the reviewer.

## Checklist

**Bureaucracy**

- [ ] I set a descriptive title
- [ ] I set a proper description, providing enough information to the reviewer
- [ ] I considered squashing this PR, if needed
- [ ] I set the proper assignee
- [ ] I set the proper reviewers

**Quality Assurance**

- [ ] I tested the behavior of this PR locally and made sure it complies with
  the ticket description

### CODE REVIEW

**Review by Toumi-Elyes — APPROVED** [review]

File: packages/service-backend/src/stripe/stripe.controller.ts [hunk 1]
```diff
@@ -7,16 +7,18 @@ import {
...
   Body,
+  Delete,
 } from '@nestjs/common';
-import { ApiOkResponse, ApiQuery, ApiTags } from '@nestjs/swagger';
+import { ApiBody, ApiOkResponse, ApiQuery, ApiTags } from '@nestjs/swagger';
 import { StripeService } from './stripe.service';
-import { CurrencyType } from './stripe.dto';
+import { CurrencyType, CreateProductDto } from './stripe.dto';
 import { LoggedMiddleware } from 'middleware/middleware.decorator';
+import { OllContext } from '../context/context.decorator';
```
Comment (Toumi-Elyes) [inline packages/service-backend/src/stripe/stripe.controller.ts]:
```suggestion
import { OllContext } from 'context/context.decorator';
```

File: packages/service-backend/src/stripe/stripe.service.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import { Injectable } from '@nestjs/common';
 import Stripe from 'stripe';
 import { STRIPE_PRIVATE_KEY, STRIPE_WEBHOOK_SECRET } from 'setup';
-import { CurrencyType } from './stripe.dto';
+import { CreateProductDto, CurrencyType } from './stripe.dto';
+import prisma from '../client';
```
Comment (Toumi-Elyes) [inline packages/service-backend/src/stripe/stripe.service.ts]:
```suggestion
import prisma from 'client';
```

File: packages/service-backend/package.json [hunk 3]
```diff
@@ -73,7 +73,7 @@
     "jest": "^29.4.3",
     "jest-junit": "^16.0.0",
     "prettier": "^2.8.4",
-    "prisma": "5.15.0",
+    "prisma": "^5.19.1",
```
Comment (Toumi-Elyes) [inline packages/service-backend/package.json:76]: Is there a particular reason why we should upgrade prisma?

**Review by Toumi-Elyes — APPROVED** [review]

File: packages/service-backend/src/stripe/stripe.controller.ts [hunk 4]
```diff
@@ -102,4 +102,16 @@ export class StripeController {
   ) {
     return await this.stripeService.createPaymentLink(product_id, ctx);
   }
+
+  @ApiOkResponse({
+    description: 'Session status checked successfully',
+  })
```
Comment (Toumi-Elyes) [inline packages/service-backend/src/stripe/stripe.controller.ts:108]: Is there a specific return type?

File: packages/service-backend/package.json (hunk 3 above)
Comment (ThisisYoYoDev) [reply packages/service-backend/package.json:76]: Nope! just the new version :)

**Pull request merged**